docs/interaction_based_testing.adoc

```diff
@@ -1056,6 +1056,68 @@ mock.type == List
 mock.nature == MockNature.MOCK
 ----
 
+[[DetachedMockFactory]]
+=== Create Mocks Outside Specifications
+
+Sometimes, it can be helpful to create and possibly preconfigure mock, stub or spy objects outside specifications, especially if you want to factor out duplicate or similar code from there.
+Use `DetachedMockFactory` to create such mock objects.
+
+NOTE: Detached mocks, as the name implies, must be attached to a specification to make them functional.
+      This either happens manually by calling `MockUtil.attachMock(Object, Specification)` or automatically by declaring the mock as an instance field with an `@AutoAttach` annotation, see <<extensions.adoc#AutoAttach,AutoAttach>>.
+      While auto-attached mocks will also be auto-detached during specification clean-up, you need to take care of manually attached mocks by yourself, calling `MockUtil.detachMock(Object)` when they are no longer in use.
+
+Assuming that we have an application class `Car` and each car has an `Engine`:
+
+[source,groovy,indent=0]
+----
+include::{sourcedir}/interaction/DetachedMockFactoryDocSpec.groovy[tags=engine;car]
+----
+
+In our specification, we declare a detached mock factory and a mock util, either inside a feature method or, if we need them for multiple features, as shared or static instances:
+
+[source,groovy,indent=0]
+----
+include::{sourcedir}/interaction/DetachedMockFactoryDocSpec.groovy[tags=declare-static]
+----
+
+==== Manually attach / detach
+
+Now in our feature, we create a detached `Engine` mock, stub its `isStarted()` method, attach the mock to the specification manually, inject the mock into a `Car` subject under specification, use the subject and finally detach the mock again after use:
+
+[source,groovy,indent=0]
+----
+include::{sourcedir}/interaction/DetachedMockFactoryDocSpec.groovy[tags=attach-manually]
+----
+
+==== Use @AutoAttach
+
+Same situation, different approach: We let Spock take care of automatically attaching the detached mock when starting the feature method and detaching it again after running the feature.
+
+[source,groovy,indent=0]
+----
+include::{sourcedir}/interaction/DetachedMockFactoryDocSpec.groovy[tags=auto-attach]
+----
+
+==== Pre-configure detached mock with default response
+
+This advanced use case might only be of interest to a minority of users, you may skip it for now if you just want to learn Spock basics.
+
+Usually, you would simply create your detached mock outside of Spock, but then define interactions (stubbed method results, method call verifications) inside the feature method itself. Sometimes however, you might want to define some (user-overridable) default behaviour right in the detached mock definition itself and even make it somewhat configurable. This is possible using a combination of Spock's mocking API and custom `IDefaultResponse` implementations.
```

Comment:
I think the primary, use-case of this would be to make the mock respond in a meaningful manner outside the context of a specification. For example if you mock something in spring and the bean is immediately used by other beans without the chance to get attached first.

And add a new line for every sentence in asciidoc, it makes diffs easier to read.

```suggestion
Usually, you would simply create your detached mock outside of Spock, but then define interactions (stubbed method results, method call verifications) inside the feature method itself.
Sometimes however, you might want to define some (user-overridable) default behaviour right in the detached mock definition itself and even make it somewhat configurable.
This is possible using a combination of Spock's mocking API and custom `IDefaultResponse` implementations.
```

Reply:
Good point, I will adjust the explanation.